parlai/core/params.py

```diff
@@ -638,6 +832,24 @@ def parse_known_args(self, args=None, namespace=None, nohelp=False):
             args = [a for a in args if a != '-h' and a != '--help']
         return super().parse_known_args(args, namespace)
 
+    def _load_known_opts(self, optfile, parsed):
+        with open(optfile, 'r', encoding='utf-8') as handle:
+            new_opt = json.load(handle)
+        for key, value in new_opt.items():
+            # existing commandline parameters take priority.
+            if key not in parsed or parsed[key] is None:
+                parsed[key] = value
+
+    def _load_opts(self, opt):
+        optfile = opt.get('init_opt')
+        with open(optfile, 'r', encoding='utf-8') as handle:
+            new_opt = json.load(handle)
+        for key, value in new_opt.items():
+            # existing commandline parameters take priority.
+            if key not in self.opt['override']:
+                self.opt[key] = value
```

Comment:
quick question: you pass in opt as an arg here, so why are you calling `self.opt` instead of `opt`?

Reply:
fixed, they're actually the same anyway rn..